analyzers/src/SonarAnalyzer.CSharp/Rules/LoggingArgumentsShouldBePassedCorrectly.cs

```diff
@@ -123,4 +124,7 @@ private static void CheckInvalidTypeParams(InvocationExpressionSyntax invocation
 
     private static bool IsInvalidArgument(ArgumentSyntax argumentSyntax, SemanticModel model, ImmutableArray<KnownType> knownTypes) =>
         model.GetTypeInfo(argumentSyntax.Expression).Type?.DerivesFromAny(knownTypes) is true;
+
+    private static ImmutableArray<KnownType> Filter(IMethodSymbol methodSymbol, ImmutableArray<KnownType> knownTypes) =>
+        knownTypes.Where(knownType => !methodSymbol.ConstructedFrom.Parameters.Any(x => x.Type.DerivesFrom(knownType))).ToImmutableArray();
```

Comment:
Is there any point in making the result of this to `ImmutableArray`?

Why not `ToArray`? I assume constructing an `ImmutableArray` is more expensive, and there is no race condition here (it's all happening on the same thread.